### PR TITLE
Scoring

### PR DESCRIPTION
**Pull request by ian-travers**
Since our server does not track the reputation of the players, there are no ratings or stars of the week.

This pull request, with partial caveats, implements it all.

Added two new server options:
- BanRoomsCreation - prohibits the creation of your own rooms.
- BanCheater - prohibits players with a cheat from entering rooms.

### CODE REVIEW

File: nfsuserver/nfsuserver2.cpp [hunk 1]
```diff
@@ -63,10 +66,7135 @@ bool running = true;
...
+};
+
+void ShiftPersona(UserClass* user, RegUser* tr, int l) {
+	strcpy(user->Personas[l], user->Personas[l + 1]);
+	memset(user->Personas[l + 1], 0, 16);
+	strcpy(tr->Personas[l], tr->Personas[l + 1]);
+	memset(tr->Personas[l + 1], 0, 16);
+};
+
+bool IsCheater(const unsigned char wei, const unsigned char sus, const unsigned char eng, const unsigned char tur,
```
Comment (HarpyWar) [inline nfsuserver/nfsuserver2.cpp:190]: Could you describe in simple words how this function works?

Comment (HarpyWar) [inline nfsuserver/nfsuserver2.cpp:190]: Could you describe in simple words how this function works?

**Comment by HarpyWar** [comment]
Is there incompatibility with old `rusers.dat` file format?

File: nfsuserver/nfsuserver2.cpp (hunk 1 above)
Comment (ian-travers) [reply nfsuserver/nfsuserver2.cpp:190]: I'll try to be briefly.
A player can get only 3 TJ's parts of 9 when completing a career mode. This function determines whether a set of TJ's parts is valid if a player had complete the career. The player is a cheater when:

- more than 3 TJ's parts
- doesn't have one of first set of TJ's parts (engine, transmission, tires)
- has more than one of the third set of TJ's pars (weight reduction, suspension, NOS)

**Comment by ian-travers** [comment]
> Is there incompatibility with old `rusers.dat` file format?

Yes, there is. 16 instead of 100 bytes per username and each persona.

`
struct RegUser {
	char Username[16];
	char Personas[4][16];
	RegUser* Next;
};
`

**Pull request merged**